libs/config/src/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+set(headers
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/asio.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/autolink.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/compiler_fence.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/compiler_specific.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/debug.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/export_definitions.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/lambda_capture.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/version.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/warnings_suffix.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/attributes.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/branch_hints.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/compiler_native_tls.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/constexpr.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/emulate_deleted.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/forceinline.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/manual_profiling.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/threads_stack.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/warnings_prefix.hpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx/config/weak_symbol.hpp
+)
+
+set(sources
+  ${CMAKE_CURRENT_SOURCE_DIR}/version.cpp
+)
+
+add_library(hpx.config ${hpx_libs_link_mode} ${sources})
+
+target_link_libraries(hpx.config hpx.pp)
+target_include_directories(hpx.config PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>)
+
+include(HPX_AddSourceGroup)
+add_hpx_source_group(NAME hpx.config ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx
+  CLASS "Header Files" TARGETS ${headers})
+add_hpx_source_group(NAME hpx.config ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../include/hpx
+  CLASS "Source Files" TARGETS ${sources})
+
```

Comment:
Ok, I see the source_group definitions now, please (partially) disregard my previous comment. Unfortunately, this will not properly define the source groups as 

a) the `source_group` invocation has to happen in the same directory as the corresponding `add_library` is located
b) all files listed in a source group must be explicitly added to that `add_library`

Reply:
@hkaiser could you clarify the problem. To my eyes both of those conditions are fulfilled above, so I must be missing something.

Reply:
@msimberg no, unfortunately none of those conditions are met. 

a) The `source_group` call needs to happen in the same cmake scope as the corresponding `add_library` the files of the source group are part of (see Brad's answer here: https://gitlab.kitware.com/cmake/cmake/issues/18856)

b) The files added to the source group have to be passed to the corresponding `add_library` call. Currently, the files are not directly passed to the `add_library` invocation, see: https://github.com/STEllAR-GROUP/hpx/blob/master/src/CMakeLists.txt#L229-L244.

The best is to have a look at how we add our Natvis files (the VS debugger visualizers) to the generated VS solution:
![image](https://user-images.githubusercontent.com/568063/56430957-c6feaa00-628d-11e9-9e37-bbb44ef28263.png)
That happens here: https://github.com/STEllAR-GROUP/hpx/blob/master/src/CMakeLists.txt#L88-L91.